Automattic-Tracks-iOSTests/Test Helpers/Mocks.swift

```diff
@@ -9,12 +9,13 @@ enum MockError: Error {
     case generic
 }
 
-struct MockEventLoggingDataSource: EventLoggingDataSource {
-    let loggingEncryptionKey: String
-    let previousSessionLogPath: URL?
-    var currentSessionLogPath: URL?
-    let logUploadURL: URL
-    let logUploadQueueStorageURL: URL
+class MockEventLoggingDataSource: EventLoggingDataSource {
```

Comment:
This object was getting super heavy trying to do everything in the constructor. Using private setters effectively makes the object immutable, and the individual mutator methods are much more straightforward this way.

Reply:
I am not sure if I fully understand the issue here. The way we normally utilize this in Kotlin is to rely on the default values for the arguments and only pass what should be different for a test case. It looks like this would also work in Swift as such:
```
MockEventLoggingDataSource(logUploadUrl: url)
```

Am I missing something?

For the record, I am not against your current implementation, I just want to better understand the issue you're trying to solve.

Reply:
Unfortunately, Swift doesn't support argument elision like this (and likely never will, though I can't find the post about that just now) – even though the arguments are named it's a more classical approach where they have to be in order. 

If I add a new method / property to the `EventLogginDataSource`, I have to put it in the mock constructor, which is quickly getting out of hand in tests (where I typically only want to customize one field at a time). That's why I figured I'd use this design - it keeps the `MockEventLoggingDataSource` code a lot shorter, as well as the tests (because they don't have to pass every argument if they don't need them)